File: Basic Question Bank/a054/a054.java
import java.util.Scanner;

public class a054 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String[] map = new String[]{"BNZ", "AMW", "KLY", "JVX", "HU", "GT", "FS", "ER", "DOQ", "CIP"};
        char[] arr = sc.nextLine().toCharArray();
        int sum = 0;
        for(int i = 0; i < arr.length - 1; i++) {
            sum += ((int)(arr[i] - '0')) * (8 - i);
        }
        sum += (int)(arr[arr.length - 1] - '0');
        int target = (10 - (sum % 10)) % 10;
        System.out.println(map[target]);
        sc.close();
    }
}   

File: Basic Question Bank/a054/a054.py
string = str(input())
arr = ["BNZ", "AMW", "KLY", "JVX", "HU", "GT", "FS", "ER", "DOQ", "CIP"]
sum = 0
for i in range(0, len(string) - 1):
    sum += (ord(string[i]) - ord('0')) * (8 - i)
sum += ord(string[len(string) - 1]) - ord('0')
if (10 - (sum % 10) % 10) == 10:
    print("BNZ")
else:
    print(arr[(10 - (sum % 10) % 10)])